lib/graphql/stitching/util.rb

```diff
@@ -50,6 +50,14 @@ def self.get_possible_types(schema, parent_type)
       def self.is_leaf_type?(type)
         type.kind.scalar? || type.kind.enum?
       end
+
+      def self.get_named_type_for_field_node(schema, parent_type, node)
+        if node.name == "__schema" && parent_type == schema.query
+          schema.types["__Schema"] # type mapped to phantom introspection field
```

Comment:
This is such a regrettable hack... but alas, it's better than hacking the `GraphQL::Schema` object to provide the data.